lib/Driver/Driver.cpp

```diff
@@ -1481,14 +1481,21 @@ void Driver::buildOutputInfo(const ToolChain &TC, const DerivedArgList &Args,
     // top-level output.
     OI.ShouldGenerateModule = true;
     OI.ShouldTreatModuleAsTopLevelOutput = true;
-  } else if ((OI.DebugInfoLevel > IRGenDebugInfoLevel::LineTables &&
-              OI.shouldLink()) ||
-             Args.hasArg(options::OPT_emit_objc_header,
-                         options::OPT_emit_objc_header_path)) {
+  } else if (OI.DebugInfoLevel > IRGenDebugInfoLevel::LineTables &&
+             OI.shouldLink()) {
     // An option has been passed which requires a module, but the user hasn't
     // requested one. Generate a module, but treat it as an intermediate output.
     OI.ShouldGenerateModule = true;
     OI.ShouldTreatModuleAsTopLevelOutput = false;
+  } else if (Args.hasArg(options::OPT_emit_objc_header,
+                         options::OPT_emit_objc_header_path,
+                         options::OPT_experimental_emit_interface) &&
+             OI.CompilerMode != OutputInfo::Mode::SingleCompile) {
```

Comment:
Why the additional `OI.CompilerMode != OutputInfo::Mode::SingleCompile` condition here?